core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -847,7 +846,12 @@ void handleJoinRequest(final DiscoveryNode node, final MembershipAction.JoinCall
 
             // validate the join request, will throw a failure if it fails, which will get back to the
             // node calling the join request
-            membership.sendValidateJoinRequestBlocking(node, joinTimeout);
+            try {
+                membership.sendValidateJoinRequestBlocking(node, state, joinTimeout);
+            } catch (Throwable e) {
+                callback.onFailure(new IllegalStateException("Can't handle join request from node [" + node + "], probable cause missing plugins", ExceptionsHelper.unwrapCause(e)));
```

Comment:
I changed the message as suggested in https://github.com/areek/elasticsearch/commit/3cc23f001c9345f666740b2deacb5033cf7b9a58